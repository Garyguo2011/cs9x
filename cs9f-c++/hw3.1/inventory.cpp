#include <iostream>
#include "inventory.h"

using namespace std;
bool QuantityComparer (Entry& a, Entry& b);

Inventory::Inventory() {}
void Inventory::Update(string item, int amount) {
	map<string,int>::iterator it;
	it = table.find(item);
	if (it == table.end()){
		table[item] = amount;
	} else {
		table[item] += amount;
	}
	cout << item << " :  " << table[item] << endl;
}

void Inventory::ListByName (){
	for(auto iter = table.begin(); iter != table.end(); iter++) {
		cout << iter->first << " :  " << iter->second << endl;
	}
}

void Inventory::ListByQuantity() {
	vector<Entry> v;
	for(auto iter = table.begin(); iter != table.end(); iter++) {
		v.push_back(*iter);
	}
	sort(v.begin(), v.end(), QuantityComparer);
	for(auto iter = v.begin(); iter != v.end(); iter++) {
		cout << iter->first << " :  " << iter->second << endl;	
	}
}

bool QuantityComparer (Entry& a, Entry& b){
	return a.second < b.second;
}
