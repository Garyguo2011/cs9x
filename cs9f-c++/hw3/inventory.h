#include <vector>
#include <string>
// #include <utility>
#include <map>

using namespace std;
typedef pair<string, int> Entry;

class Inventory {
	public:
		Inventory ();
		void Update (string item, int amount);
		void ListByName ();
		void ListByQuantity ();
	private:
		// You provide this.
		map<string, int> table;
};
