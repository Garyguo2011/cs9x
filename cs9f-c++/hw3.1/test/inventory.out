apple :  2
orange :  4
Unrecognizable command word.
apple :  -2
apple :  -2
orange :  4
Unrecognizable command args.
apple :  -2
orange :  4
